getting-started/mbed_profile.json

```diff
@@ -0,0 +1,16 @@
+{
+    "ARMC6": {
+        "common": ["-c", "--target=arm-arm-none-eabi", "-mthumb", "-O1",
```

Comment:
Running the build with `mbed compile -vv …` shows that if this is used to override the default profile, the compiler command line combines options from the two profile files. Repeating options here is not only superfluous, it's misleading, since for example if we wanted to avoid a `-DFOO` in the default profile we'd need to have `-UFOO` here. Therefore this should just be `"common": ["-Os"]`. The same goes for IAR.

Alternatively, if this profile is meant to be used on its own, it needs to have the settings for the other tools (`"c"`, `"cxx"`, `"ld"`). Otherwise the build fails.